saleor/graphql/order/types.py

```diff
@@ -184,14 +185,20 @@ class Meta:
     @staticmethod
     @traced_resolver
     def resolve_user(root: models.OrderEvent, info):
-        user = info.context.user
-        if (
-            user == root.user
-            or user.has_perm(AccountPermissions.MANAGE_USERS)
-            or user.has_perm(AccountPermissions.MANAGE_STAFF)
-        ):
-            return root.user
-        raise PermissionDenied()
+        def _resolve_user(event_user):
+            user = info.context.user
+            if (
+                user == event_user
+                or user.has_perm(AccountPermissions.MANAGE_USERS)
+                or user.has_perm(AccountPermissions.MANAGE_STAFF)
```

Comment:
In all user resolvers, we should use similar logic to `saleor/saleor/graphql/account/resolvers.py::resolve_user` 
We should check if the requested user is staff and then check proper permission.